x-pack/plugins/alerting/server/alert/alert.ts

```diff
@@ -172,6 +176,46 @@ export class Alert<
     return {
       state: this.state,
       meta: this.meta,
+      flappingHistory: this.flappingHistory,
+    };
+  }
+
+  toRawRecovered(): RawAlertRecoveredInstance {
+    return {
+      flappingHistory: this.flappingHistory,
+    };
+  }
+
+  setFlappingHistory(fh: boolean[]) {
+    this.flappingHistory = fh;
+  }
+
+  getFlappingHistory() {
+    return this.flappingHistory;
+  }
+
+  updateFlappingHistory(state: boolean) {
+    const { atCapacity, diff } = this.flappingHistoryAtCapacity();
+    if (atCapacity) {
+      this.flappingHistory = drop(this.flappingHistory, diff);
+    }
+    this.flappingHistory.push(state);
+  }
+
+  isFlapping(): boolean {
```

Comment:
When an alert is created should the default flapping state be true or false? Currently it will be set to false.

Reply:
Heh, I keep getting confused on this - I think it's wording.  The flapping "history" should end up being ["true"] for a newly created alert, since it will then be considered active.  But the alert won't be considered "flapping" at this point, since this is the first time it's been seen.

Reply:
After talking, I think we are going to keep it as false for now, but we would appreciate any input from the team on this is no longer in draft.

Reply:
Removed some of the alerting functions in this commit 2326d577e517965b3a9c6271d026c38326d2c23f